libs/books/feature/src/lib/book-search/book-search.component.spec.ts

```diff
@@ -1,27 +1,74 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import {
+  async,
+  ComponentFixture,
+  TestBed,
+  fakeAsync,
+  tick
+} from '@angular/core/testing';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
 import { SharedTestingModule } from '@tmo/shared/testing';
 
 import { BooksFeatureModule } from '../books-feature.module';
 import { BookSearchComponent } from './book-search.component';
+import { clearSearch, searchBooks } from '@tmo/books/data-access';
+import { MockStore, provideMockStore } from '@ngrx/store/testing';
 
 describe('ProductsListComponent', () => {
   let component: BookSearchComponent;
   let fixture: ComponentFixture<BookSearchComponent>;
+  let store: MockStore;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      imports: [BooksFeatureModule, NoopAnimationsModule, SharedTestingModule]
+      imports: [BooksFeatureModule, NoopAnimationsModule, SharedTestingModule],
+      providers: [provideMockStore()]
     }).compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(BookSearchComponent);
     component = fixture.componentInstance;
+    store = TestBed.inject(MockStore);
+    jest.spyOn(store, 'dispatch');
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeDefined();
   });
-});
+
+  describe('Handle search term change', () => {
+    it('should dispatch searchBooks action when not empty as per  after 500ms debounceTime', fakeAsync(() => {
```

Comment:
Correct the description. It should match what you are testing.
Applicable wherever this kind of description present in this file.